classes/local/object_manipulator/manipulator_builder.php

```diff
@@ -0,0 +1,112 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * File manipulator abstract class.
+ *
+ * @package   tool_objectfs
+ * @author    Gleimer Mora <gleimermora@catalyst-au.net>
+ * @copyright Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace tool_objectfs\local\object_manipulator;
+
+use coding_exception;
+use dml_exception;
+use moodle_exception;
+use stdClass;
+use tool_objectfs\local\object_manipulator\candidates\candidates_finder;
+use tool_objectfs\log\aggregate_logger;
+
+defined('MOODLE_INTERNAL') || die();
+
+class manipulator_builder {
```

Comment:
I Feel like this class is trying to do two different things  - both build the manipulators and execute them, i'd rather one of two things happen

1. We have a separate build class that builds the objects and they are executed by another
2. Only expose the execute and execute_all methods as public and have the building be implicit within them and not called separately.

Reply:
I chose the option 2 :smile: